### PR TITLE
[Go-Map] Update ProjectBuilder & Tests 2: Electric Boogalloo

### PR DESCRIPTION
**Pull request by Lewiky**
Same as before, except this time all of the e2e tests have been removed to help minimise the diff

### CODE REVIEW

File: src/test/resources/go/complex-user/dao/dao.go.snippet [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+	"github.com/squat/and/dab/complexuser/util"
+	"github.com/google/uuid"
+
+	// pq acts as the driver for SQL requests
+	_ "github.com/lib/pq"
+)
+
+// Datastore provides the interface adopted by the DAO, allowing for mocking
+type Datastore interface {
+	CreateComplexuser(input CreateComplexuserInput) (*Complexuser, error)
```
Comment (jaylees14) [inline src/test/resources/go/complex-user/dao/dao.go.snippet:17]: Don't think this is you, probably @smithwjv - could we make this `ComplexUser`?

**Review by jaylees14 — APPROVED** [review]
All looks good to me - obviously can't run the Go files, but a quick read and everything looks sensible.

Do wish we supported `SELECT * FROM table` tho 😝

File: src/test/resources/go/complex-user/dao/dao.go.snippet (hunk 1 above)
Comment (willjvsmith) [reply src/test/resources/go/complex-user/dao/dao.go.snippet:17]: Should be handled if it comes in as `complexUser`

Comment (willjvsmith) [reply src/test/resources/go/complex-user/dao/dao.go.snippet:17]: It capitalises whatever comes in for the service name

**Review by c-harding — CHANGES_REQUESTED** [review]
*Boogaloo

File: src/test/resources/go/complex-user/dao/dao.go.snippet (hunk 1 above)
Comment (c-harding) [reply src/test/resources/go/complex-user/dao/dao.go.snippet:17]: This is a coding error, nowhere in the input does it appear with a lowercase `u`.

File: src/main/scala/temple/Main.scala [hunk 2]
```diff
@@ -13,7 +15,7 @@ object Main extends App {
   try {
     val config = new TempleConfig(args)
     config.subcommand match {
-      case Some(config.Generate) => Application.generate(config)
+      case Some(config.Generate) => Application.generate(config, PoliceSergeantNicholasAngel)
```
Comment (c-harding) [inline src/main/scala/temple/Main.scala:18]: I still think this should be a mock and not the main application

**Review by c-harding — APPROVED** [review]

File: src/main/scala/temple/Main.scala (hunk 2 above)
Comment (Lewiky) [reply src/main/scala/temple/Main.scala:18]: That was gonna come in a separate PR

**Pull request merged**